modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+                ADReal miuox = (qwR[a] * miubalibx_norm) + (qxR[a] * (0.0)) +
+                               (qyR[a] * miubalibz_norm) - (qzR[a] * miubaliby_norm);
+                ADReal miuoy = (qwR[a] * miubaliby_norm) - (qxR[a] * miubalibz_norm) +
+                               (qyR[a] * (0.0)) + (qzR[a] * miubalibx_norm);
+                ADReal miuoz = (qwR[a] * miubalibz_norm) + (qxR[a] * miubaliby_norm) -
+                               (qyR[a] * miubalibx_norm) + (qzR[a] * (0.0));
+
+                // Determine whether the boundary normal is taken from the miinima library or
+                // compuuted from orientation quaterniionss
+                ADReal miubalibwR = 0.0;
+                ADReal miubalibxR = miubalibx_norm;
+                ADReal miubalibyR = miubaliby_norm;
+                ADReal miubalibzR = miubalibz_norm;
+
+                if (_BoundaryNormal == 0)
+                {
+                  miubalibwR = ((miuow) * (qwR[a])) - ((miuox) * (qxR[a]) * (-1)) -
+                               ((miuoy) * (qyR[a]) * (-1)) - ((miuoz) * (qzR[a]) * (-1));
+                  miubalibxR = ((miuow) * (qxR[a]) * (-1)) + ((miuox) * (qwR[a])) +
+                               ((miuoy) * (qzR[a]) * (-1)) - ((miuoz) * (qyR[a]) * (-1));
+                  miubalibyR = ((miuow) * (qyR[a]) * (-1)) - ((miuox) * (qzR[a]) * (-1)) +
+                               ((miuoy) * (qwR[a])) + ((miuoz) * (qxR[a]) * (-1));
+                  miubalibzR = ((miuow) * (qzR[a]) * (-1)) + ((miuox) * (qyR[a]) * (-1)) -
+                               ((miuoy) * (qxR[a]) * (-1)) + ((miuoz) * (qwR[a]));
+                }
+
+                _VwlibR[_qp] = miubalibwR;
+                _VxlibR[_qp] = miubalibxR;
+                _VylibR[_qp] = miubalibyR;
+                _VzlibR[_qp] = miubalibzR;
+
+                if (sumval == 0.0)
+                {
+                  miubalibwR = 0.0;
+                  miubalibxR = 0.0;
+                  miubalibyR = 0.0;
+                  miubalibzR = 0.0;
+
+                  _VwlibR[_qp] = 0.0;
+                  _VxlibR[_qp] = 0.0;
+                  _VylibR[_qp] = 0.0;
+                  _VzlibR[_qp] = 0.0;
+                }
+
+                // dot product of normalized anisotropic vector and normalized gradient vector
+
+                ADReal dot_product =
+                    (miubalibxR * vsmallbax) + (miubalibyR * vsmallbay) + (miubalibzR * vsmallbaz);
+
+                // calculate the switch for a specific pair of grains and each library
+                // misorientation
+
+                if ((thetabalib[l]) < 0)
+                {
+                  (thetabalib[l]) = 6.28319 + (thetabalib[l]);
+                }
+
+                ADReal thetadtheta = thetaba - (thetabalib[l]);
+
+                ADReal normvbavalue = (vbax * vbax) + (vbay * vbay) + (vbaz * vbaz);
+
+                ADReal normvba = std::pow(normvbavalue, .5);
```

Comment:
```suggestion
                ADReal normvba = std::sqrt(normvbavalue);
```
always better to use the specific operators `std::sqrt` and `std:cbrt`. pow is generally much slower.

Reply:
See updated files